openapi-ls-extension/src/main/java/io/ballerina/openapi/extension/OpenAPIConverterService.java

```diff
@@ -100,30 +112,96 @@ public OpenAPIConverterService() {
+            for (DocumentId currentDocumentID : defaultModule.documentIds()) {
+                Document document = defaultModule.document(currentDocumentID);
+
+                SyntaxTree syntaxTree = document.syntaxTree();
+                List<OASResult> oasResults = ServiceToOpenAPIConverterUtils.generateOAS3Definition(
+                        syntaxTree, semanticModel.get(), null, false, null);
+
+                generateServiceJson(response, document.syntaxTree().filePath(), oasResults, specs);
             }
+            response.setContent(specs);
             return response;
         });
     }
 
+    /**
+     * Generate openAPI json for a service.
+     *
+     * @param response     OpenAPIConverterResponse to set error message
+     * @param document     Ballerina document path
+     * @param oasResults   OAS Results list
+     * @param specs        Specs array to add the service
+     */
+    private void generateServiceJson(OpenAPIConverterResponse response, String document,
+                                     List<OASResult> oasResults, JsonArray specs) {
+        for (OASResult oasResult : oasResults) {
+            if (oasResult.getOpenAPI().isEmpty()) {
+                response.setError("Error occurred while generating yaml.");
+                continue;
+            }
+            if (oasResult.getJson().isEmpty()) {
+                continue;
+            }
+            JsonObject spec = new JsonObject();
+            JsonElement json = JsonParser.parseString(oasResult.getJson().get()).getAsJsonObject();
+            JsonArray diagnosticsJson = getDiagnosticsJson(oasResult);
+
+            spec.addProperty(SERVICE_NAME, oasResult.getOpenAPI().get().getInfo().getTitle());
+            spec.addProperty(FILE, document);
+            spec.add(SPEC, json);
+            spec.add(DIAGNOSTICS, diagnosticsJson);
+            specs.add(spec);
+        }
+    }
+
+    /**
+     * Generate and returns the diagnostics data in OAS Result.
+     *
+     * @param oasResult OASResult
+     * @return Json Array of diagnostics
+     */
+    private JsonArray getDiagnosticsJson(OASResult oasResult) {
+        List<OpenAPIConverterDiagnostic> diagnostics = oasResult.getDiagnostics();
+        JsonArray diagnosticsJson = new JsonArray();
+        for (OpenAPIConverterDiagnostic diagnostic : diagnostics) {
+            JsonObject diagnosticJson = new JsonObject();
+            diagnosticJson.addProperty(MESSAGE, diagnostic.getMessage());
+            diagnosticJson.addProperty(SEVERITY, diagnostic.getDiagnosticSeverity().name());
+            Optional<Location> diagnosticLocation = diagnostic.getLocation();
+            if (diagnosticLocation.isPresent()) {
+                GsonBuilder builder = new GsonBuilder();
+                builder.serializeNulls();
+                Gson gson = builder.create();
+                diagnosticJson.add(LOCATION, gson.toJsonTree(diagnosticLocation.get().lineRange()));
```

Comment:
What are the nullable fields which can be in the `LineRange`?

Reply:
There are situations with handling some exceptions during the code generator process in the tool since it is hard to catch the location for exceptions we pass the `null` as value for the location to the diagnostic `OpenAPIConverterDiagnostic` model.